Author: Thomas Koch <thomas@koch.ro>
Description: Adapt ant build.xml to Debian
 - take dependency libs from /usr/share/java
 - do not run svn
 - read debian/ant.properties
 - generate protobuf files
 - do not compile rhino from fork in lib/rhino (not in +dfsg orig tarball)
 - the new artifact version scheme after the move to git is vYEARMMDD
 - remove build-only dependencies from pom
Last-Update: 2013-02-22
Forwarded: not-needed

--- a/build.xml
+++ b/build.xml
@@ -76,18 +76,20 @@
   <property file="build.properties" />
 
   <!-- maven ant tasks -->
-  <path id="maven-ant-tasks.classpath" path="${tools.dir}/maven-ant-tasks-2.1.3.jar" />
+  <path id="maven-ant-tasks.classpath" path="${lib.dir}/maven-ant-tasks.jar" />
   <typedef resource="org/apache/maven/artifact/ant/antlib.xml"
            uri="antlib:org.apache.maven.artifact.ant"
            classpathref="maven-ant-tasks.classpath" />
 
   <!-- gather svn version -->
   <target name="svnversion">
+<!--
     <exec outputproperty="build.svnVersion"
       executable="svnversion"
       failonerror="false"
       failifexecutionfails="false"
       dir="."/>
+-->
   </target>
 
   <!-- compile rhino -->
@@ -102,13 +104,14 @@
   </target>
 
   <target name="rhino-jarjar"
-          depends="rhino"
+          depends=""
           description="Renamespaces Rhino">
+    <mkdir dir="${buildlib.dir}" />
     <taskdef name="jarjar"
              classname="com.tonicsystems.jarjar.JarJarTask"
-             classpath="lib/jarjar.jar"/>
+             classpath="${lib.dir}/jarjar.jar"/>
     <jarjar destfile="${buildlib.dir}/rhino.jar" update="true">
-      <zipfileset src="${buildlib.dir}/rhino1_7R5pre/js.jar"/>
+      <zipfileset src="${lib.dir}/js.jar"/>
       <rule pattern="org.mozilla.javascript.**"
             result="com.google.javascript.rhino.head.@1"/>
     </jarjar>
@@ -119,6 +122,7 @@
       <include name="**/*.proto"/>
     </fileset>
     <pathconvert property="protofiles" pathsep=" " refid="proto.classpath"/>
+    <mkdir dir="${gen.dir}"/>
     <echo message="${protoc.executable} -I ${src.dir} --java_out=${gen.dir} ${protofiles}"/>
     <exec executable="${protoc.executable}" searchpath="true">
       <arg line="-I ${src.dir}"/>
@@ -136,9 +140,9 @@
       </filterset>
     </copy>
 
-    <property name="compiler-jarfile-nodeps" value="${build.dir}/closure-${ant.project.name}-r${build.svnVersion}.jar" />
-    <property name="compiler-jarfile-javadoc" value="${build.dir}/closure-${ant.project.name}-r${build.svnVersion}-javadoc.jar" />
-    <property name="compiler-jarfile-sources" value="${build.dir}/closure-${ant.project.name}-r${build.svnVersion}-sources.jar" />
+    <property name="compiler-jarfile-nodeps" value="${build.dir}/closure-${ant.project.name}-v${build.svnVersion}.jar" />
+    <property name="compiler-jarfile-javadoc" value="${build.dir}/closure-${ant.project.name}-v${build.svnVersion}-javadoc.jar" />
+    <property name="compiler-jarfile-sources" value="${build.dir}/closure-${ant.project.name}-v${build.svnVersion}-sources.jar" />
 
     <artifact:pom id="project" file="${build.dir}/pom.xml" />
   </target>
@@ -224,7 +228,7 @@
       <include name="guava.jar"/>
       <include name="json.jar"/>
       <include name="jsr305.jar"/>
-      <include name="protobuf-java.jar"/>
+      <include name="protobuf.jar"/>
     </fileset>
     <fileset dir="${buildlib.dir}">
       <include name="rhino.jar"/>
@@ -234,7 +238,12 @@
   <path id="allclasspath.path">
     <pathelement location="${classes.dir}" />
     <fileset dir="${lib.dir}">
-      <include name="*.jar"/>
+      <include name="junit4.jar"/>
+      <include name="args4j.jar"/>
+      <include name="guava.jar"/>
+      <include name="json.jar"/>
+      <include name="jsr305.jar"/>
+      <include name="protobuf.jar"/>
     </fileset>
     <fileset dir="${buildlib.dir}">
       <include name="rhino.jar"/>
@@ -243,11 +252,12 @@
 
   <target name="clean" description="delete generated files">
     <delete dir="${build.dir}" />
+    <delete dir="${gen.dir}" />
   </target>
 
   <target name="compile"
           description="compile the source code"
-          depends="rhino-jarjar,svnversion">
+          depends="rhino-jarjar, protobuf-gen">
     <mkdir dir="${classes.dir}" />
     <javac srcdir="${gen.dir}"
            destdir="${classes.dir}"
@@ -304,14 +314,6 @@
     <jar destfile="${compiler-jarfile}" update="true">
       <fileset dir="${classes.dir}" />
       <fileset dir="${build.dir}" includes="externs.zip" />
-      <zipfileset src="${lib.dir}/args4j.jar"/>
-      <zipfileset src="${lib.dir}/guava.jar"/>
-      <zipfileset src="${lib.dir}/json.jar"/>
-      <zipfileset src="${lib.dir}/jsr305.jar"/>
-      <zipfileset src="${lib.dir}/protobuf-java.jar"/>
-
-      <zipfileset src="${buildlib.dir}/rhino.jar"/>
-
       <manifest>
         <attribute name="Main-Class"
                    value="com.google.javascript.jscomp.CommandLineRunner" />
--- a/closure-compiler.pom
+++ b/closure-compiler.pom
@@ -24,7 +24,7 @@
   <packaging>jar</packaging>
 
   <name>Closure Compiler</name>
-  <version>r@build.svnVersion@</version>
+  <version>v@build.svnVersion@</version>
 
   <url>http://code.google.com/p/closure-compiler/</url>
   <description>
@@ -140,26 +140,12 @@
     </dependency>
 
     <dependency>
-      <groupId>org.apache.ant</groupId>
-      <artifactId>ant</artifactId>
-      <version>1.8.2</version>
-      <scope>compile</scope>
-    </dependency>
-
-    <dependency>
       <groupId>com.google.code.findbugs</groupId>
       <artifactId>jsr305</artifactId>
       <version>1.3.9</version>
     </dependency>
 
     <dependency>
-      <groupId>com.googlecode.jarjar</groupId>
-      <artifactId>jarjar</artifactId>
-      <version>1.1</version>
-      <scope>compile</scope>
-    </dependency>
-
-    <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
       <version>4.10</version>
